Backend/utils/db_utils.py

```diff
@@ -0,0 +1,220 @@
+
+def createTableFromFile(settings_file, dbInstance):
+  try:
+    with open(settings_file) as f:
+      data = f.read()
+      settings = json.loads(data)
+
+      attribute_definitions = settings.get("AttributeDefinitions")
+      key_schema = settings.get("KeySchema")
+      provisioned_throughput = settings.get("ProvisionedThroughput")
+      table_name = settings.get("TableName")
+
+      # check for required parameters
+      if table_name is not None and key_schema is not None and attribute_definitions is not None and provisioned_throughput is not None:
+        if len(key_schema) > 0 and len(attribute_definitions) > 0:
+          table = dbInstance.create_table(
+              AttributeDefinitions=attribute_definitions,
+              KeySchema=key_schema,
+              ProvisionedThroughput=provisioned_throughput,
+              TableName=table_name
+            )
+          f.close()
+          return table
+      f.close()
+  except IOError as e:
+    message = "Unable to create table from file"
+    logger.exception(message + ': ' + settings_file + '\n' + str(e))
+    return None
+
+def createTableFromJson(settings_json, dbInstance):
+  if (type(settings_json) == str):
+    settings = json.loads(settings_json)
+  else:
+    settings = settings_json
+
+  attribute_definitions = settings.get("AttributeDefinitions")
+  key_schema = settings.get("KeySchema")
+  provisioned_throughput = settings.get("ProvisionedThroughput")
+  table_name = settings.get("TableName")
+
+  # check for required parameters
+  if table_name is not None and key_schema is not None and attribute_definitions is not None and provisioned_throughput is not None:
+    if len(key_schema) > 0 and len(attribute_definitions) > 0:
+      table = dbInstance.create_table(
+          AttributeDefinitions=attribute_definitions,
+          KeySchema=key_schema,
+          ProvisionedThroughput=provisioned_throughput,
+          TableName=table_name
+        )
+      return table
+  message = "Unable to create table from json"
+  logger.error(message)
+  return None
+
+def preloadDataFromFile(fileName, table):
+  try:
+    with open(fileName) as json_file:
+      data = json_file.read()
+      items = json.loads(data)
+      for item in items:
+        if "email" in item.keys() and "password" in item.keys() and "activationId" in item.keys() and "active" in item.keys():
+          email = item['email']
+          password = item['password']
+          activationId = item['activationId']
+          active = item['active']
+
+          table.put_item(
+            Item={
+               'email': email,
+               'password': password,
+               'activationId': activationId,
+               'active': active
+            }
+          )
+        else:
+          message = "Unable to add item to table " + table.describe_table + ". Missing email, password, activationId, or active status"
+          logger.error(message)
+      json_file.close()
+  except IOError as e:
```

Comment:
This exception is handled by using the with statement. It happens in a few places. See the last code snippit in this section. https://docs.python.org/2/tutorial/inputoutput.html#methods-of-file-objects